homeassistant/components/purpleair/__init__.py

```diff
@@ -71,6 +75,23 @@ def __init__(
             ATTR_LONGITUDE: self.sensor_data.longitude,
         }
 
+    @property
+    def extra_state_attributes(self) -> Mapping[str, Any]:
+        """Return entity specific state attributes."""
+        attrs = {}
+
+        # Displaying the geography on the map relies upon putting the latitude/longitude
+        # in the entity attributes with "latitude" and "longitude" as the keys.
+        # Conversely, we can hide the location on the map by using other keys, like
+        # "lati" and "long":
+        if self._entry.options.get(CONF_SHOW_ON_MAP):
+            attrs[ATTR_LATITUDE] = self.sensor_data.latitude
+            attrs[ATTR_LONGITUDE] = self.sensor_data.longitude
+        else:
+            attrs["lati"] = self.sensor_data.latitude
+            attrs["long"] = self.sensor_data.longitude
```

Comment:
Why do se set state attributes at all in this case?

Reply:
Because that information may still be useful for a user who wants to know the precise location of a sensor (but doesn't want that on the map).

Reply:
I have seen multiple PRs in related to `show on map`, I would not hold this PR, but it worth making an architecture issue to add a flag which the entity will be able to set if it wants to hide icons but still expose coordinates.